File: socialuni-js/packages/socialuni-expand/socialuni-expand-api/src/api/SocialuniUserLikeAPI.ts
import SocialuniUserExtendFriendQueryQO from "socialuni-api-base/src/model/user/SocialuniUserExtendFriendQueryQO";
import CenterUserDetailRO from "socialuni-api-base/src/model/social/CenterUserDetailRO";
import socialuniUserRequest from "socialuni-user-api/src/request/socialuniUserRequest";
import SocialuniUserIdQO from "socialuni-api-base/src/model/QO/user/SocialuniUserIdQO";
import SocialuniUserExtendDetailRO from "../model/SocialuniUserExtendDetailRO";
import SocialuniLikeAllConfigBO from "../model/SocialuniLikeAllConfigBO";
import MessageAddVO from "socialuni-im-api/src/model/QO/message/MessageAddVO";
import socialuniImRequest from "socialuni-im-api/src/api/socialuniImRequest";
import MessageVO from "socialuni-im-api/src/model/RO/MessageVO";

export default class SocialuniUserLikeAPI {

  static addUserLikeAPI(user: SocialuniUserExtendDetailRO) {
    return socialuniUserRequest.post<CenterUserDetailRO[]>('socialuni/userLike/likeUser', {userId: user.id})
  }

  static getAllConfig() {
    return socialuniUserRequest.get<SocialuniLikeAllConfigBO>('socialuni/userLike/getLikeAllConfig')
  }

  static sendMsgAPI(chatId: string, content: string, type: string = null) {
    const msgAdd: MessageAddVO = new MessageAddVO(chatId, content, type)
    return socialuniImRequest.post <MessageVO>('socialuni/userLike/sendMsg', msgAdd)
  }
}

File: socialuni-server/socialuni-likee-sdk/src/main/java/com/socialuni/social/likee/api/SocialuniUserLikeAPI.java
package com.socialuni.social.likee.api;

import com.socialuni.social.common.api.model.ResultRO;
import com.socialuni.social.common.api.model.user.SocialuniUserIdQO;
import com.socialuni.social.im.api.model.QO.SocialuniChatQueryQO;
import com.socialuni.social.im.api.model.QO.message.MessageAddVO;
import com.socialuni.social.im.api.model.RO.ChatRO;
import com.socialuni.social.im.api.model.RO.SocialMessageRO;
import com.socialuni.social.likee.model.SocialuniLikeAllConfigBO;
import com.socialuni.social.likee.model.SocialuniLikeChatRO;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

@FeignClient(name = "userLike", url = "${socialuni.central-server-url:https://api.socialuni.cn}", path = "socialuni/userLike")
@Tag(name = "用户模块/喜欢用户")
public interface SocialuniUserLikeAPI {
    @PostMapping("likeUser")
    @Operation(summary = "喜欢用户")
    ResultRO<Void> likeUser(@RequestBody @Valid SocialuniUserIdQO addVO);

    @GetMapping("getLikeAllConfig")
    @Operation(summary = "得到喜欢域的所有配置")
    ResultRO<SocialuniLikeAllConfigBO> getLikeAllConfig();

    @PostMapping("sendMsg")
    ResultRO<SocialMessageRO> sendMsg(@RequestBody @Valid MessageAddVO messageAddVO);

    @PostMapping("queryChat")
    ResultRO<SocialuniLikeChatRO> queryChat(@RequestBody @Valid @NotNull SocialuniChatQueryQO socialuniChatQueryQO);

}